crates/solidity/testing/snapshots/cst_output/InterfaceDefinition/sample_counter/generated/0.4.11.yml

```diff
@@ -15,44 +15,46 @@ Tree:
   - InterfaceDefinition (Rule): # 0..172 "interface ICounter {\n    // returns the current co..."
       - Interface (Token): "interface" # 0..9
       - Identifier (Token): "ICounter" # 10..18
-      - Group: # 19..172 "{\n    // returns the current count\n    function co..."
+      - _DELIMITEDBY (Rule): # 18..172 " {\n    // returns the current count\n    function c..."
           - OpenBrace (Token): "{" # 19..20
-          - Group: # 25..170 "// returns the current count\n    function count() ..."
-              - ContractBodyElement (Rule): # 25..104 "// returns the current count\n    function count() ..."
-                  - FunctionDefinition (Rule): # 25..104 "// returns the current count\n    function count() ..."
+          - _REPEATED (Rule): # 21..171 "    // returns the current count\n    function coun..."
+              - ContractBodyElement (Rule): # 21..106 "    // returns the current count\n    function coun..."
+                  - FunctionDefinition (Rule): # 21..106 "    // returns the current count\n    function coun..."
                       - Function (Token): # 25..66 "// returns the current count\n    function"
                           - SingleLineComment (Trivia): "// returns the current count" # 25..53
                           - Contents: "function" # 58..66
                       - Identifier (Token): "count" # 67..72
                       - ParameterList (Rule): # 72..74 "()"
-                          - OpenParen (Token): "(" # 72..73
-                          - CloseParen (Token): ")" # 73..74
-                      - Group: # 75..88 "external view"
+                          - _DELIMITEDBY (Rule): # 72..74 "()"
+                              - OpenParen (Token): "(" # 72..73
+                              - CloseParen (Token): ")" # 73..74
+                      - _REPEATED (Rule): # 75..88 "external view"
                           - FunctionAttribute (Rule): # 75..83 "external"
                               - External (Token): "external" # 75..83
                           - FunctionAttribute (Rule): # 84..88 "view"
                               - View (Token): "view" # 84..88
-                      - Group: # 89..103 "returns (uint)"
-                          - Returns (Token): "returns" # 89..96
-                          - ParameterList (Rule): # 97..103 "(uint)"
+                      - Returns (Token): "returns" # 89..96
```

Comment:
Another instance of the issue where the CST shape changes based on whether an `optional` has a value or not. See the one that holds the `returns` token.

Reply:
This means that to find out whether a function is returning a value or not, the user won't be able to access it by a known index `func.children[X]` , or even its kind `func.children.find(|x| x.kind = ReturnValue)`.

Instead, they will be searching for one or more of its children, that might exist at random locations, or even more error-prone, by the kind of said children. In this specific case, there are two `ParameterList` in the parent. No indication of which is which.

Reply:
If you want a node for the a sequence you have to name it in the grammar. Otherwise it will be hoisted.

Reply:
Not hoisting anonymous sequences leads to very messy trees, because they are often non-structural to deal with parsing combinator precedence issues.

Reply:
And optionals don't generate a wrapping node, for the same overly verbose reason.

Reply:
I see. Thanks for pointing out. Naming them will give it the correct shape then. That resolves the above concern.